### PR TITLE
Test enum utilities

### LINKED ISSUE #586: [Testing] `packages/website/src/shared/lib/utilities/enum.ts`
### User Story

As a developer, I want the codebase to be fully tested, so that I am sure that we are promoting high code quality by releasing stable code with no regressions.

### Description

This work, along with many other tickets, is to improve our testing code coverage by writing tests for the file in the title.

To view our current code coverage, run `pnpm test`. Then, open this file: `<project-root>/coverage/lcov-report/index.html`.

### Acceptance Criteria

1. The file in the title is fully tested, including statements, branches, functions, and lines.
2. All existing test suites pass.

### Technical References

- [Jest documentation](https://jestjs.io/docs/getting-started)
- [React Testing Library documentation](https://testing-library.com/docs/react-testing-library/intro/)
- Look into existing test suites

### Additional Information

N/A

### PR DESCRIPTION
**Pull request by glenflorendo**
### Description

This PR provides complete testing of the enum utility functions.

```
Enum Utilities
    getNth
      ✓ gets the nth enum (2 ms)
      ✓ returns undefined for out-of-bounds nth
    getFirst
      ✓ gets first enum
      ✓ returns undefined for empty enum
```

### Related Issues

Resolves #586

### Testing

- Ran existing testing suites
- Wrote new tests

```
-----------------------------------------------|---------|----------|---------|---------|-----------------------
File                                           | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s     
-----------------------------------------------|---------|----------|---------|---------|-----------------------
 src/shared/lib/utilities                      |   24.13 |        0 |   33.33 |      15 |                       
  enum.ts                                      |     100 |      100 |     100 |     100 |                       
```

### Checklist

- [x] I have followed all conventions outlined in our documentation.
- [x] I have fully tested my changes and confirmed that all new and existing tests pass.
- [x] I have written meaningful commit messages for all changes.
- [x] I have linted and formatted my changes to follow the code style of this repository.
- [x] I have updated our documentation, accordingly.
- [x] I have checked currently opened pull requests to ensure that there are no pending pull request for the same changes or issues.
- [x] I have confirmed that this pull request fully meets the acceptance criteria for all related issues listed above.

### CODE REVIEW

File: packages/website/src/shared/lib/utilities/enum.test.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  LOL = "League of Legends",
+  OW = "Overwatch",
+  VAL = "Valorant",
+}
+
+enum MOCK_EMPTY_ENUM {}
+
+describe("Enum Utilities", () => {
+  describe("getNth", () => {
+    it("gets the nth enum", () => {
```
Comment (gibsonliketheguitar) [inline packages/website/src/shared/lib/utilities/enum.test.ts:18]: Not sure if this is intended effect. Tried some things out on coderpad with codebelow

```
import { chain, nth, values, each, size, range } from "lodash";

export const getNth = (enums: Record<any, any>, index: number) => chain(enums).values().nth(index).value();

enum Test {
  a = 1,
  b = 2,
  c = 3
}

console.log(Test)
// { '1': 'a' , '2': 'b', '3': 'c', a: 1, b:2, c:3}


each(range(0, size(Test)), (n) => {
  console.log(n)
});

Output:
0
1
2
3
4
5
6

```

Found out, [enums transpiling number values](https://stackoverflow.com/questions/56044872/typescript-enum-object-values-return-value) by Typescript creates 2x size of enum. While test passes currently, the above quirk is like an edge case.

File: packages/website/src/shared/lib/utilities/enum.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import _ from "lodash";
 
-export const getNthEnum = (enums: Record<any, any>, index: number) => _.chain(enums).values().nth(index).value();
+export const getNth = (enums: Record<any, any>, index: number) => _.chain(enums).values().nth(index).value();
```
Comment (gibsonliketheguitar) [inline packages/website/src/shared/lib/utilities/enum.ts:4]: Maybe we can do something more vanillia?


```
function getNth(enums: Record<any, any>, index: number){
  const values = Object.values(enums)
  
  if(index.length > values.length) return undefined
  return values[index]
}

```

**Review by gibsonliketheguitar — CHANGES_REQUESTED** [review]

File: packages/website/src/shared/lib/utilities/enum.test.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import _ from "lodash";
```
[the remark this answers is not in the document]
Comment (glenflorendo) [reply packages/website/src/shared/lib/utilities/enum.test.ts:1]: Definitely not needed, but I rather we stay consistent. Also, eventually, we won't include our test files in the bundle, so I'm okay with convenience if it helps test thoroughly.

File: packages/website/src/shared/lib/utilities/enum.ts (hunk 2 above)
Comment (glenflorendo) [reply packages/website/src/shared/lib/utilities/enum.ts:4]: Yeah, `lodash` is definitely just a convenience library. There's nothing it provies we can't do ourselves. Is there a reason why you rather not use it?

File: packages/website/src/shared/lib/utilities/enum.test.ts (hunk 1 above)
Comment (glenflorendo) [reply packages/website/src/shared/lib/utilities/enum.test.ts:18]: That's good to keep note of. Though, I don't think this happens with the test, since we use string enums.

Mock enum output:
```
console.log(MOCK_ENUM);

console.log
    {
      FF: 'Final Fantasy',
      FNAF: "Five Nights at Freddy's",
      GTA: 'Grand Theft Auto',
      KH: 'Kingdom Hearts',
      LOL: 'League of Legends',
      OW: 'Overwatch',
      VAL: 'Valorant'
    }
```

Test with log:
```
_.each(_.range(0, _.size(MOCK_ENUM)), (n) => {
  console.log("n:", n, "Received:", getNth(MOCK_ENUM, n), "Expected:", _.nth(_.toArray(MOCK_ENUM), n));
  expect(getNth(MOCK_ENUM, n)).toStrictEqual(_.nth(_.toArray(MOCK_ENUM), n));
});
```

Test output:
```
    n: 0 Received: Final Fantasy Expected: Final Fantasy
    n: 1 Received: Five Nights at Freddy's Expected: Five Nights at Freddy's
    n: 2 Received: Grand Theft Auto Expected: Grand Theft Auto
    n: 3 Received: Kingdom Hearts Expected: Kingdom Hearts
    n: 4 Received: League of Legends Expected: League of Legends
    n: 5 Received: Overwatch Expected: Overwatch
    n: 6 Received: Valorant Expected: Valorant
```

**Review by gibsonliketheguitar — APPROVED** [review]

**Comment by glenflorendo** [comment]
Fixed merge conflicts.
